src/Sulu/Bundle/AdminBundle/Resources/js/components/Select/OverlayList.js

```diff
@@ -0,0 +1,150 @@
+import Option from './Option';
+import overlayListStyles from './overlayList.scss';
+import type {OverlayListDimensions} from './types';
+import OverlayListPositioner from './OverlayListPositioner';
+
+@observer
+export default class OverlayList extends React.PureComponent {
+    props: {
+        isOpen: boolean,
+        children?: React.Element<*>,
+        onRequestClose?: () => void,
+        /** The top coordinate relative to which the list will be positioned **/
+        anchorTop: number,
+        /** The left coordinate relative to which the list will be positioned **/
+        anchorLeft: number,
+        /** The width of the element relative to which the list will be positioned **/
+        anchorWidth: number,
+        /** The width of the element relative to which the list will be positioned **/
+        anchorHeight: number,
+        /** The index of the child element which will be centered relative to the anchor **/
+        centeredChildIndex: number,
+    };
+
+    static defaultProps = {
+        isOpen: false,
+        centeredChildIndex: 0,
+        anchorTop: 0,
+        anchorLeft: 0,
+        anchorWidth: 0,
+        anchorHeight: 0,
+    };
+
+    @observable offsetHeight: number;
+    @observable offsetWidth: number;
+    @observable centeredChildRelativeTop: number;
+    list: HTMLElement;
+    scrollTop: number;
+
+    componentDidMount() {
+        window.addEventListener('blur', this.requestClose);
+        window.addEventListener('resize', this.requestClose);
+    }
+
+    componentWillUnmount() {
+        window.removeEventListener('blur', this.requestClose);
+        window.removeEventListener('resize', this.requestClose);
+    }
+
+    componentDidUpdate() {
+        window.requestAnimationFrame(() => {
+            if (this.list) {
+                this.list.scrollTop = this.scrollTop;
+            }
+        });
+    }
+
+    requestClose = () => {
+        if (this.props.isOpen && this.props.onRequestClose) {
+            this.props.onRequestClose();
+        }
+    };
+
+    @computed get dimensions(): ?OverlayListDimensions {
+        if (!this.offsetHeight || !this.offsetHeight || !this.centeredChildRelativeTop) {
+            return null;
+        }
+        const positioner = new OverlayListPositioner(
+            this.offsetHeight,
+            this.offsetWidth,
+            this.centeredChildRelativeTop,
+            this.props.anchorTop,
+            this.props.anchorLeft,
+            this.props.anchorWidth,
+            this.props.anchorHeight,
+        );
+        return positioner.getCroppedDimensions();
+    }
+
+    readCenteredChildRelativeTop = (option: Option) => {
+        if (option) {
+            window.requestAnimationFrame(action(() => {
+                this.centeredChildRelativeTop = option.getOffsetTop();
+            }));
+        }
+    };
+
+    readOffsetDimensions = (list: HTMLElement) => {
+        this.list = list;
+        if (!list || this.offsetWidth || this.offsetHeight) {
```

Comment:
Yes this should be the case. To calculate the correct position and size of the list, the width and height of the list before resizing is needed. Maybe we could rename the variables to make that more clear. What about `naturalOffsetWidth/Height`